Plugins/Sycl/include/Acts/Plugins/Sycl/Seeding/Seedfinder.ipp

```diff
@@ -0,0 +1,145 @@
+  m_config.minHelixDiameter2 =
+      std::pow(m_config.minPt * 2 / m_config.pTPerHelixRadius, 2);
+  m_config.pT2perRadius =
+      std::pow(m_config.highland / m_config.pTPerHelixRadius, 2);
+}
+
+template <typename external_spacepoint_t>
+Seedfinder<external_spacepoint_t>::Seedfinder(
+    Acts::SeedfinderConfig<external_spacepoint_t> config,
+    const Acts::Sycl::DeviceExperimentCuts& cuts)
+    : m_config(config),
+      m_seedFilterConfig(m_config.seedFilter->getSeedFilterConfig()),
+      m_deviceCuts(cuts),
+      m_wrappedQueue() {
+  init();
+}
+
+template <typename external_spacepoint_t>
+Seedfinder<external_spacepoint_t>::Seedfinder(
+    Acts::SeedfinderConfig<external_spacepoint_t> config,
+    const Acts::Sycl::DeviceExperimentCuts& cuts,
+    Acts::Sycl::QueueWrapper&& wrappedQueue)
+    : m_config(config),
+      m_seedFilterConfig(m_config.seedFilter->getSeedFilterConfig()),
+      m_deviceCuts(cuts),
+      m_wrappedQueue(std::move(wrappedQueue)) {
+  init();
+}
+
+template <typename external_spacepoint_t>
+Seedfinder<external_spacepoint_t>::Seedfinder(
+    Acts::SeedfinderConfig<external_spacepoint_t> config,
+    const Acts::Sycl::DeviceExperimentCuts& cuts,
+    const Acts::Sycl::QueueWrapper& wrappedQueue)
+    : m_config(config),
+      m_seedFilterConfig(m_config.seedFilter->getSeedFilterConfig()),
+      m_deviceCuts(cuts),
+      m_wrappedQueue(wrappedQueue) {
+  init();
+}
+
+template <typename external_spacepoint_t>
+template <typename sp_range_t>
+std::vector<Acts::Seed<external_spacepoint_t>>
+Seedfinder<external_spacepoint_t>::createSeedsForGroup(
+    sp_range_t bottomSPs, sp_range_t middleSPs, sp_range_t topSPs) const {
+  std::vector<Seed<external_spacepoint_t>> outputVec;
+
+  std::vector<detail::DeviceSpacePoint> deviceBottomSPs;
+  std::vector<detail::DeviceSpacePoint> deviceMiddleSPs;
+  std::vector<detail::DeviceSpacePoint> deviceTopSPs;
+
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*>
+      bottomSPvec;
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*>
+      middleSPvec;
+  std::vector<const Acts::InternalSpacePoint<external_spacepoint_t>*> topSPvec;
+
+  for (auto SP : bottomSPs) {
+    bottomSPvec.push_back(SP);
+    deviceBottomSPs.push_back(
+        detail::DeviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                                 SP->varianceR(), SP->varianceZ()});
+  }
+
+  for (auto SP : middleSPs) {
+    middleSPvec.push_back(SP);
+    deviceMiddleSPs.push_back(
+        detail::DeviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                                 SP->varianceR(), SP->varianceZ()});
+  }
+
+  for (auto SP : topSPs) {
+    topSPvec.push_back(SP);
+    deviceTopSPs.push_back(
+        detail::DeviceSpacePoint{SP->x(), SP->y(), SP->z(), SP->radius(),
+                                 SP->varianceR(), SP->varianceZ()});
+  }
+
+  const auto deviceConfigData = detail::DeviceSeedfinderConfig{
```

Comment:
My comment about this variable was also closed without changing the code. :confused: Could you make this a member variable on the class?

Reply:
I only looked at the four bottom lines, and thought I know what you're talking about. Turns out I didn't.

Reply:
Should be fine now. :)